openml/extensions/sklearn/extension.py

```diff
@@ -128,19 +136,44 @@ def remove_all_in_parentheses(string: str) -> str:
                 string, removals = re.subn(r"\([^()]*\)", "", string)
             return string
 
-        name = long_name
-        if not name.startswith('sklearn'):
-            raise ValueError("Expected 'sklearn' in as start of flow name. "
-                             "Make sure a sklearn-flow is provided.")
-        if 'sklearn.model_selection' in name and not name.startswith('sklearn.model_selection'):
-            raise ValueError("Model Selection is not outer scope. "
-                             "This is unexpected, create a new issue with the flow id.")
-        short_name = 'sklearn.{}'
+        # Generally, we want to trim all hyperparameters, the exception to that is for model
+        # selection, as the `estimator` hyperparameter is very indicative of what is in the flow.
+        # So we first trim pipeline names of the `estimator` parameter. For reference:
```

Comment:
I'm having trouble to match

> So we first trim pipeline names of the `estimator` parameter.

with the code. The code does not really check for pipelines below, but for model selection. I think the issue is that you do not mean a scikit-learn pipeline object by pipeline, is that correct? Could you update the comment to reflect this more clearly?

Reply:
That's right. I meant any number of components specified under `estimators=` of the model selection. I understand it's confusing and I'll change it.